cmd/modern/root/config/add-user.go

```diff
@@ -18,35 +19,48 @@ import (
 type AddUser struct {
 	cmdparser.Cmd
 
-	name            string
-	authType        string
-	username        string
-	encryptPassword bool
+	name               string
+	authType           string
+	username           string
+	passwordEncryption string
 }
 
 func (c *AddUser) DefineCommand(...cmdparser.CommandOptions) {
-	options := cmdparser.CommandOptions{
-		Use:   "add-user",
-		Short: "Add a user",
-		Examples: []cmdparser.ExampleOptions{
-			{
-				Description: "Add a user (using the SQLCMD_PASSWORD environment variable)",
-				Steps: []string{
-					fmt.Sprintf(`%s SQLCMD_PASSWORD=<placeholderpassword>`, pal.CreateEnvVarKeyword()),
-					"sqlcmd config add-user --name my-user --username user1",
-					fmt.Sprintf(`%s SQLCMD_PASSWORD=`, pal.CreateEnvVarKeyword()),
-				},
+	examples := []cmdparser.ExampleOptions{
+		{
+			Description: "Add a user (using the SQLCMD_PASSWORD environment variable)",
+			Steps: []string{
+				fmt.Sprintf(`%s SQLCMD_PASSWORD=<placeholderpassword>`, pal.CreateEnvVarKeyword()),
+				"sqlcmd config add-user --name my-user --username user1 --password-encryption none",
+				fmt.Sprintf(`%s SQLCMD_PASSWORD=`, pal.CreateEnvVarKeyword()),
 			},
-			{
-				Description: "Add a user (using the SQLCMDPASSWORD environment variable)",
-				Steps: []string{
-					fmt.Sprintf(`%s SQLCMDPASSWORD=<placeholderpassword>`, pal.CreateEnvVarKeyword()),
-					"sqlcmd config add-user --name my-user --username user1",
-					fmt.Sprintf(`%s SQLCMDPASSWORD=`, pal.CreateEnvVarKeyword()),
-				},
+		},
+		{
+			Description: "Add a user (using the SQLCMDPASSWORD environment variable)",
+			Steps: []string{
+				fmt.Sprintf(`%s SQLCMDPASSWORD=<placeholderpassword>`, pal.CreateEnvVarKeyword()),
+				"sqlcmd config add-user --name my-user --username user1 --password-encryption none",
+				fmt.Sprintf(`%s SQLCMDPASSWORD=`, pal.CreateEnvVarKeyword()),
 			},
 		},
-		Run: c.run}
+	}
+
+	if runtime.GOOS == "windows" {
+		examples = append(examples, cmdparser.ExampleOptions{
+			Description: "Add a user using Windows Data Protection API to encrypt password in sqlconfig",
+			Steps: []string{
+				fmt.Sprintf(`%s SQLCMD_PASSWORD=<placeholderpassword>`, pal.CreateEnvVarKeyword()),
+				"sqlcmd config add-user --name my-user --username user1 --password-encryption dpapi",
+				fmt.Sprintf(`%s SQLCMD_PASSWORD=`, pal.CreateEnvVarKeyword()),
```

Comment:
>	fmt.Sprintf(`%s SQLCMD_PASSWORD=`, pal.CreateEnvVarKeyword()), [](http://example.com/codeflow?start=3&length=63)

it'd be useful to add a 
`func setEnvironmentVar(var string, val string) string` to `pal` so you don't have to use fmt.Sprintf everywhere and to handle any potential need to escape the value in a per-platform manner

Reply:
Added (so we don't lose this feedback in this merged PR):

https://github.com/microsoft/go-sqlcmd/issues/307